### PR TITLE
feat: Expose git commit id in the health check endpoint

### PR DESCRIPTION
**Pull request by KShivendu**
### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

This will help with identifying commit ID in our CI Grafana dashboard.

```
➜  ~ curl localhost:6333
{"title":"qdrant - vector search engine","version":"0.11.1","commit":"64ec1149956f298c54b9f11e765949a0e141565e"}%                                  

➜  ~ grpcurl -plaintext localhost:6334 qdrant.Qdrant/HealthCheck
{
  "title": "qdrant - vector search engine",
  "version": "0.11.1",
  "commit": "64ec1149956f298c54b9f11e765949a0e141565e"
}
```

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [x] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by KShivendu** [comment]
@generall please review

File: lib/api/src/grpc/models.rs [hunk 1]
```diff
@@ -4,17 +4,20 @@ use schemars::JsonSchema;
 use serde;
 use serde::{Deserialize, Serialize};
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, JsonSchema)]
 pub struct VersionInfo {
     pub title: String,
     pub version: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub commit_id: Option<String>,
```
Comment (timvisee) [inline lib/api/src/grpc/models.rs:12]: Nitpick: I believe this is commonly called `commit`, would that be better?

**Review by timvisee — APPROVED** [review]
Awesome work! :rocket: 

As a minor addition to this I quickly made it optional. This prevents `git` and a repository from being a requirement to build Qdrant. That may be useful for packaging in some cases, for example.

File: lib/api/src/grpc/models.rs (hunk 1 above)
Comment (KShivendu) [reply lib/api/src/grpc/models.rs:12]: I was thinking the same. Will do. Thanks!

**Comment by timvisee** [comment]
It appears that my `optional` changes broke compilation somehow, but I'm a bit clueless on why this suddenly popped up. :sweat_smile:

**Comment by KShivendu** [comment]
So as per [probuf.dev](https://protobuf.dev/programming-guides/field_presence/#how-to-enable-explicit-presence-in-proto3), proto3 doesn't allow that by default. 

I tried this and it fixed the error:
```rust
tonic_build::configure()
        .configure_validation()
        .file_descriptor_set_path(build_out_dir.join("qdrant_descriptor.bin"))
        .protoc_arg("--experimental_allow_proto3_optional") // --> this
        .out_dir("src/grpc/")
        .compile("...")
....
```

However, I'm sceptical of doing this because it's "experimental" and it would probably force us to use the same flag in all of our SDKs. I think it's better to just use an empty string if git isn't present while building. What do you think? 

cc @generall

**Comment by KShivendu** [comment]
Ohh looks like the newer versions (v3.15.0) of `protoc` [supports this](https://protobuf.dev/programming-guides/field_presence/#protoc-invocation) without the experimental flag. 

I tried [v3.15.0](https://github.com/protocolbuffers/protobuf/releases/download/v3.15.0/protoc-3.15.0-linux-x86_64.zip) locally and it worked without the flag. 

This means the [setup-protoc](https://github.com/arduino/setup-protoc) GH action that we use doesn't pick up the latest versions of protoc by default.

**Comment by timvisee** [comment]
The weird thing is that we use `optional` in a lot of other `.proto` files which does work fine, so this isn't anything new.

File: openapi/tests/openapi_integration/test_ping_api.py [hunk 2]
```diff
@@ -1,14 +1,14 @@
 from .helpers.helpers import request_with_validation
 
-def test_health_api():
+def test_ping_api():
```
Comment (timvisee) [inline openapi/tests/openapi_integration/test_ping_api.py:3]: Nitpick: maybe just `test_root` or `test_root_api`.

Comment (KShivendu) [reply openapi/tests/openapi_integration/test_ping_api.py:3]: Good one. I was confused what to name it. I'll update the OpenAPI docs as well.

**Pull request merged**